freqtrade/freqtradebot.py

```diff
@@ -1236,11 +1236,16 @@ def handle_stoploss_on_exchange(self, trade: Trade) -> bool:
             self.handle_protections(trade.pair, trade.trade_direction)
             return True
 
-        if trade.open_entry_or_exit_orders_count != 0 or not trade.is_open:
-            # Trade has an open Buy or Sell order, Stoploss-handling can't happen in this case
-            # as the Amount on the exchange is tied up in another trade.
-            # The trade can be closed already (sell-order fill confirmation came in this iteration)
-            return False
+        print("***************open_orders DEBUG***************")
+        print(f"trade.open_orders: {trade.open_orders}")
+        print(f"trade.open_orders_count: {trade.open_orders_count}")
+        print(f"trade.open_entry_or_exit_orders_count: {trade.open_entry_or_exit_orders_count}")
+
+        # if trade.open_entry_or_exit_orders_count != 0 or not trade.is_open:
```

Comment:
The function is not working with that code block, From my understanding this issue of tied capital is only present on spot, can you confirm? Do you have an idea of how we can achieve this? Why not letting the exchange return an error and handling it?

Reply:
the "tied up" issue is really depending on the exchange, and how they actually implement the stop order types.

on binance, it's only spot - on futures, i think it depends if the exchange supports reduce-only orders.
if they don't - then you risk reversing your position accidentally by having both a stop and limit order open (and triggered).

i don't really see why this shouldn't be working here - we're in a trade scoped method and have the trade object available ... 
if `trade.open_entry_or_exit_orders_count` is not returning the right value, then there's a problem with that method i'd assume.

having the exchange fail it intentionally is problematic and not a solution.
it'll never succeed place a stoploss, but _attempt_ to place one at any iteration (until an eventually order tying up the capital is cancelled or closed). That'll lead to TONS of bad log messages - as well as potentially problems with exchange ratelimits.

Reply:
did you notice it in tests - or in actually running the bot?
I'm asking because with tests - it could just as well be caused by inadequate test setup.

We've improved this over time - but not all tests require 100% valid trades - so they're not always perfectly aligned.

Reply:
I've got this logs from the tests, I will check the test

Reply:
In fact pytest tests/test_freqtradebot.py::test_handle_stoploss_on_exchange is working but it is other test cases using the function that are failing, I think it could be more related to bad mock order initialisation than an issue within the function.